components/edit-collective/SocialLinksFormField.tsx

```diff
@@ -0,0 +1,270 @@
+import React from 'react';
+import { Plus, Times } from '@styled-icons/fa-solid';
+import { DragIndicator } from '@styled-icons/material';
+import { sortBy } from 'lodash';
+import { DndProvider, useDrag, useDrop } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import { FormattedMessage } from 'react-intl';
+
+import { SocialLink, SocialLinkInput, SocialLinkType } from '../../lib/graphql/types/v2/graphql';
+
+import { Box, Flex } from '../Grid';
+import StyledButton from '../StyledButton';
+import StyledInput from '../StyledInput';
+import StyledSelect from '../StyledSelect';
+import { Span } from '../Text';
+
+export type SocialLinksFormFieldProps = {
+  value?: SocialLink[];
+  onChange: (value: SocialLinkInput[]) => void;
+};
+
+export default function SocialLinksFormField({ value, onChange }: SocialLinksFormFieldProps) {
+  const onItemChange = React.useCallback(
+    (socialLink, index) => {
+      const newValues = [...value.slice(0, index), socialLink, ...value.slice(index + 1)];
+      onChange(
+        newValues.map(sl => ({
+          url: sl.url,
+          type: sl.type,
+        })),
+      );
+    },
+    [value, onChange],
+  );
+
+  const onRemoveItem = React.useCallback(
+    index => {
+      const newValues = [...value.slice(0, index), ...value.slice(index + 1)];
+      onChange(
+        newValues.map(sl => ({
+          url: sl.url,
+          type: sl.type,
+        })),
+      );
+    },
+    [value, onChange],
+  );
+
+  const addItem = React.useCallback(() => {
+    const newValues = [...value, { url: '', type: SocialLinkType.WEBSITE }];
+    onChange(
+      newValues.map(sl => ({
+        url: sl.url,
+        type: sl.type,
+      })),
+    );
+  }, [value, onChange]);
+
+  const onMoveItem = React.useCallback(
+    (fromIndex: number, toIndex: number) => {
+      const movingItem = value[fromIndex];
+      const newValues = [...value.slice(0, fromIndex), ...value.slice(fromIndex + 1)];
+      newValues.splice(toIndex, 0, movingItem);
+      onChange(
+        newValues.map(sl => ({
+          url: sl.url,
+          type: sl.type,
+        })),
+      );
+    },
+    [value, onChange],
+  );
+
+  return (
+    <DndProvider backend={HTML5Backend}>
+      <Flex p={1} flexDirection="column">
+        <Flex>
+          <Box my="5px" fontWeight={700}>
+            <FormattedMessage defaultMessage="Social Links" />
+          </Box>
+        </Flex>
+        <Flex flexDirection="column">
+          {(value || []).map((socialLink, i) => {
+            return (
+              <SocialLinkItem
+                // eslint-disable-next-line react/no-array-index-key
+                key={i}
```

Comment:
Can't indexing by `i` cause issues since we allow re-ordering? I wasn't able to spot an issue, but out of safety I would recommend:

```suggestion
                key={`${i}-{socialLink.type}-${socialLink.url}`}
```

Reply:
Keying with the url does not work well in this case, it causes unresponsive typing as the component rerenders on every key stroke of the url input.